src/main/java/com/noxag/newnox/textanalyzer/algorithms/BibliographyAnalyzer.java

```diff
@@ -1,18 +1,111 @@
 import com.noxag.newnox.textanalyzer.data.Finding;
+import com.noxag.newnox.textanalyzer.data.TextFinding;
+import com.noxag.newnox.textanalyzer.data.TextFinding.TextFindingType;
+import com.noxag.newnox.textanalyzer.data.pdf.PDFPage;
+import com.noxag.newnox.textanalyzer.data.pdf.TextPositionSequence;
+import com.noxag.newnox.textanalyzer.util.PDFTextExtractionUtil;
+import com.opencsv.CSVReader;
 
 public class BibliographyAnalyzer implements TextanalyzerAlgorithm {
+    private static final Logger LOGGER = Logger.getLogger(CommonAbbreviationAnalyzer.class.getName());
+    private static final String BIBLIOGRAPHY_IDENTIFICATION_LIST_PATH = "src/main/resources/analyzer-conf/bibliography-identifications.csv";
 
     @Override
     public List<Finding> run(PDDocument doc) {
-        // TODO Auto-generated method stub
-        return null;
+        List<Finding> findings = new ArrayList<>();
+        try {
+            findings.addAll(getReferencesWithoutSource(doc));
+        } catch (IOException e) {
+            LOGGER.log(Level.WARNING, "Could not strip text from document", e);
+        }
+        return findings;
+    }
+
+    private List<Finding> getReferencesWithoutSource(PDDocument doc) throws IOException {
+        List<Finding> findings = new ArrayList<>();
+        List<PDFPage> pages = PDFTextExtractionUtil.extractText(doc);
+        List<String> bibliographyHints = readBibliographyIdentificationListFile(BIBLIOGRAPHY_IDENTIFICATION_LIST_PATH);
+        pages.stream().forEach(page -> {
+            TextPositionSequence firstWordOfPage = page.getFirstWord();
+            if (bibliographyHints.stream().anyMatch(firstWordOfPage.toString().toLowerCase()::contains)) {
+                List<TextPositionSequence> markedSources = getAllSources(pages, firstWordOfPage.getPageIndex());
+                List<TextPositionSequence> referencedSources = getAllReferences(pages, firstWordOfPage.getPageIndex());
+                findings.addAll(compareReferencesWithSources(referencedSources, markedSources));
+            }
+        });
+        return findings;
+    }
+
+    // Gets all references to sources of content pages
+    private List<TextPositionSequence> getAllReferences(List<PDFPage> pages, int sourcePageIndex) {
+        List<PDFPage> contentPages = pages.stream().filter(page -> page.getPageIndex() < sourcePageIndex)
+                .collect(Collectors.toList());
+        return getSourceList(contentPages);
+    }
+
+    // Gets all sources in bibliography
+    private List<TextPositionSequence> getAllSources(List<PDFPage> pages, int sourcePageIndex) {
+        List<PDFPage> sourcePages = pages.stream().filter(page -> page.getPageIndex() >= sourcePageIndex)
+                .collect(Collectors.toList());
+        return getSourceList(sourcePages);
+    }
+
+    // Reads all sources/references out of given list of PDFPages
+    private List<TextPositionSequence> getSourceList(List<PDFPage> pageList) {

```

Comment:
getReferences wäre in diesem fall denke der bessere Name oder sogar getBibliographyReferences.

Reply:
done